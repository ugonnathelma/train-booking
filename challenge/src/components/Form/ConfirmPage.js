import React from "react";

import {
  Container,
  Space,
  Button,
  Label,
  FormWrap,
  Flex,
  Detail
} from "./style";

const ConfirmPage = ({
  selectedClass,
  selectedSeat,
  firstName,
  lastName,
  date,
  time,
  price,
  currentTimeLeft,
  stopTimer,
  bookingSuccess,
  cancelBooking
}) => {
  return (
    <Container column>
      <FormWrap>
        <Label>Time left to pay:</Label> <Detail time>{currentTimeLeft}</Detail>
        <Space vertical="0.8em" />
        <Label>Class:</Label> <Detail small>{selectedClass}</Detail>
        <Space vertical="0.8em" />
        <Label>Seat:</Label>
        <Detail small>{selectedSeat}</Detail>
        <Space vertical="0.8em" />
        <Label>Passenger Name:</Label>{" "}
        <Detail small>{`${firstName} ${lastName}`}</Detail>
        <Space vertical="0.8em" />
        <Label>Travel Date:</Label> <Detail small>{date}</Detail>
        <Space vertical="0.8em" />
        <Label>Travel Time:</Label> <Detail small>{time}</Detail>
        <Space vertical="0.8em" />
        <Label>Price:</Label> <Detail small>{price}</Detail>
        <Space vertical="1.5em" />
        <Flex>
          <Button
            width="10em"
            onClick={() => {
              stopTimer();
              bookingSuccess(true);
            }}
          >
            Pay
          </Button>
          <Space horizontal="1.5em" />
          <Button width="10em" onClick={() => cancelBooking()}>
            Cancel
          </Button>
        </Flex>
      </FormWrap>
    </Container>
  );
};

export default ConfirmPage;
