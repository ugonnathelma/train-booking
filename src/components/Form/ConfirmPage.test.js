import React from "react";
import ConfirmPage from "./ConfirmPage";
import { render, fireEvent } from "@testing-library/react";

describe("ConfirmPage", () => {
  const selectedSeat = 12;
  const selectedClass = 2;
  const firstName = "Tom";
  const lastName = "Riddle";
  const date = new Date("1/2/2019").getDate();
  const time = new Date("1/2/2019").getTime();
  const price = "$12";
  const currentTimeLeft = "3:12";
  const stopTimer = () => {};
  const bookingSuccess = () => {};
  const cancelBooking = () => {};

  it("should render matching the existing snapshot", () => {
    const { asFragment } = render(
      <ConfirmPage
        selectedClass={selectedClass}
        selectedSeat={selectedSeat}
        firstName={firstName}
        lastName={lastName}
        date={date}
        time={time}
        price={price}
        currentTimeLeft={currentTimeLeft}
        stopTimer={stopTimer}
        bookingSuccess={bookingSuccess}
        cancelBooking={cancelBooking}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should call cancelBooking when the cancel button is clicked", () => {
    const cancelStub = jest.fn();

    const { getByText, getByTestId } = render(
      <ConfirmPage
        selectedClass={selectedClass}
        selectedSeat={selectedSeat}
        firstName={firstName}
        lastName={lastName}
        date={date}
        time={time}
        price={price}
        currentTimeLeft={currentTimeLeft}
        stopTimer={stopTimer}
        bookingSuccess={bookingSuccess}
        cancelBooking={cancelStub}
      />
    );

    fireEvent.click(getByTestId("cancel"));

    expect(cancelStub).toHaveBeenCalled();
  });

  it("should call stopTimer and bookingSuccess when the pay button is clicked", () => {
    const stopTimerStub = jest.fn();
    const bookingSuccessStub = jest.fn();

    const { getByTestId } = render(
      <ConfirmPage
        selectedClass={selectedClass}
        selectedSeat={selectedSeat}
        firstName={firstName}
        lastName={lastName}
        date={date}
        time={time}
        price={price}
        currentTimeLeft={currentTimeLeft}
        stopTimer={stopTimerStub}
        bookingSuccess={bookingSuccessStub}
        cancelBooking={cancelBooking}
      />
    );

    fireEvent.click(getByTestId("pay"));

    expect(stopTimerStub).toHaveBeenCalled();
    expect(bookingSuccessStub).toHaveBeenCalled();
  });
});
