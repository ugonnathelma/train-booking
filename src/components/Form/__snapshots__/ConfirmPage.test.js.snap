// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPage should render matching the existing snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-bwzfXH conSAz"
  >
    <section
      class="sc-ifAKCX huQwQB"
    >
      <label
        class="sc-gqjmRU fxwXMY"
      >
        Time left to pay:
      </label>
       
      <label
        class="sc-jTzLTM leETWq"
      >
        3:12
      </label>
      <div
        class="sc-gzVnrw dRLHSk"
      />
      <label
        class="sc-gqjmRU fxwXMY"
      >
        Class:
      </label>
       
      <label
        class="sc-jTzLTM gYRASo"
      >
        2
      </label>
      <div
        class="sc-gzVnrw dRLHSk"
      />
      <label
        class="sc-gqjmRU fxwXMY"
      >
        Seat:
      </label>
      <label
        class="sc-jTzLTM gYRASo"
      >
        12
      </label>
      <div
        class="sc-gzVnrw dRLHSk"
      />
      <label
        class="sc-gqjmRU fxwXMY"
      >
        Passenger Name:
      </label>
       
      <label
        class="sc-jTzLTM gYRASo"
      >
        Tom Riddle
      </label>
      <div
        class="sc-gzVnrw dRLHSk"
      />
      <label
        class="sc-gqjmRU fxwXMY"
      >
        Travel Date:
      </label>
       
      <label
        class="sc-jTzLTM gYRASo"
      >
        2
      </label>
      <div
        class="sc-gzVnrw dRLHSk"
      />
      <label
        class="sc-gqjmRU fxwXMY"
      >
        Travel Time:
      </label>
       
      <label
        class="sc-jTzLTM gYRASo"
      >
        1546383600000
      </label>
      <div
        class="sc-gzVnrw dRLHSk"
      />
      <label
        class="sc-gqjmRU fxwXMY"
      >
        Price:
      </label>
       
      <label
        class="sc-jTzLTM gYRASo"
      >
        $12
      </label>
      <div
        class="sc-gzVnrw jrKBVA"
      />
      <section
        class="sc-EHOje cIxLkv"
      >
        <button
          class="sc-iwsKbI fxuKIw"
          data-testid="pay"
          width="10em"
        >
          Pay
        </button>
        <div
          class="sc-gzVnrw hvsyRJ"
        />
        <button
          class="sc-iwsKbI fxuKIw"
          data-testid="cancel"
          width="10em"
        >
          Cancel
        </button>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
